_overviews/core/collections-migration-213.md

```diff
@@ -0,0 +1,330 @@
+
+    Type inference has been improved so that `Map(1 -> "a").map(x => (x._1 + 1, x._2))` works, the compiler can infer the parameter type for the function literal. However, using a method reference in 2.13.0-M4 (improvement are on the way for 2.13.0) does not work, and an explicit eta-expansion is necessary:
+
+        scala> def f(t: (Int, String)) = (t._1 + 1, t._2)
+        scala> Map(1 -> "a").map(f)
+                                ^
+              error: missing argument list for method f
+              Unapplied methods are only converted to functions when a function type is expected.
+              You can make this conversion explicit by writing `f _` or `f(_)` instead of `f`.
+        scala> Map(1 -> "a").map(f _)
+        res10: scala.collection.immutable.Map[Int,String] = ChampHashMap(2 -> a)
+  - `View`s have been completely redesigned and we expect their usage to have a more predictable evaluation model.
+    You can read more about the new design [here](http://scala-lang.org/blog/2017/11/28/view-based-collections.html).
+  - `mutable.ArraySeq` (which wraps an `Array[AnyRef]` in 2.12, meaning that primitives were boxed in the array) can now wrap boxed and unboxed arrays. `mutable.ArraySeq` in 2.13 is in fact equivalent to `WrappedArray` in 2.12, there are specialized subclasses for primitive arrays. Note that a `mutable.ArraySeq` can be used either way for primitive arrays (TODO: document how). `WrappedArray` is deprecated.
+  - There is no “default” `Factory` (previously known as `[A, C] => CanBuildFrom[Nothing, A, C]`): use `Factory[A, Vector[A]]` explicitly instead.
+
+## Breaking changes with old syntax still supported
+
+The following table lists the changes that continue to work with a deprecation warning.
+
+| Description | Old Code | New Code | Automatic Migration Rule |
+| ----------- | -------- | -------- | ------------------------ |
+| `collection.Set/Map` no longer have `+` and `-` operations | `xs + 1 - 2` | `xs ++ Set(1) -- Set(2)` | `Experimental` |
+| `collection.Map` no longer have `--` operation | `map -- keys` | `map.to(immutable.Map) -- keys` | |
+| `immutable.Set/Map`: the `+` operation no longer has an overload accepting multiple values | `Set(1) + (2, 3)` | `Set(1) + 2 + 3` | `NewCollections`, `CrossCompat` |
+| `mutable.Set/Map` no longer have a `+` operation | `mutable.Set(1) + 2` | `mutable.Set(1).clone() += 2` | `NewCollections`, `CrossCompat` |
+| `SortedSet`: the `to`, `until` and `from` methods are now called `rangeTo`, `rangeUntil` and `rangeFrom`, respectively | `xs.until(42)` | `xs.rangeUntil(42)` |  |
+| `Traversable` and `TraversableOnce` are replaced with `Iterable` and `IterableOnce`, respectively | `def f(xs: Traversable[Int]): Unit` | `def f(xs: Iterable[Int]): Unit` | `NewCollections`, `CrossCompat` |
+| `Stream` is replaced with `LazyList` | `Stream.from(1)` | `LazyList.from(1)` | `RoughlyStreamToLazyList` |
```

Comment:
The new name of the rewrite rule is `Collection213Roughly`

Reply:
👍